libglusterfs: drop core infrastructure of memory pools

dmantipov opened the pull request:
Separate memory allocation type tracking from core infrastructure of
memory pools and drop the latter alongside with '--disable-mempool'
configuration option, provide '--disable-allocation-tracking' to control
the former instead.

Signed-off-by: Dmitry Antipov dantipov@cloudlinux.com
Updates: #1000

Commit:
libglusterfs: drop core infrastructure of memory pools

Separate memory allocation type tracking from core infrastructure of
memory pools and drop the latter alongside with '--disable-mempool'
configuration option, provide '--disable-allocation-tracking' to control
the former instead.

Signed-off-by: Dmitry Antipov dantipov@cloudlinux.com
Updates: #1000

gluster-ant commented:
CLANG-FORMAT FAILURE:
Before merging the patch, this diff needs to be considered for passing clang-format

```diff --git a/glusterfsd/src/glusterfsd.c b/glusterfsd/src/glusterfsd.c
index 5dca9f3f2..290e19dc1 100644
--- a/glusterfsd/src/glusterfsd.c
+++ b/glusterfsd/src/glusterfsd.c
@@ -1399,7 +1399,8 @@ parse_opts(int key, char *arg, struct argp_state *state)
         case ARGP_IO_ENGINE_KEY:
             cmd_args->io_engine = gf_strdup(arg);
             if (cmd_args->io_engine == NULL) {
-                argp_failure(state, -1, 0, "Failed to allocate memory for "
+                argp_failure(state, -1, 0,
+                             "Failed to allocate memory for "
                              "io-engine");
             }
             break;
@@ -2664,10 +2665,8 @@ out:
 int
 main(int argc, char *argv[])
 {
-    gf_io_handlers_t main_handlers = {
-        .setup = main_start,
-        .cleanup = main_terminate
-    };
+    gf_io_handlers_t main_handlers = {.setup = main_start,
+                                      .cleanup = main_terminate};
     glusterfs_ctx_t *ctx = NULL;
     int ret = -1;
     char cmdlinestr[PATH_MAX] = {
diff --git a/libglusterfs/src/xlator.c b/libglusterfs/src/xlator.c
index d00cf4872..c09ee2650 100644
--- a/libglusterfs/src/xlator.c
+++ b/libglusterfs/src/xlator.c
@@ -720,7 +720,7 @@ xlator_mem_acct_init(xlator_t *xl, int num_types)
 {
 #ifdef GF_DISABLE_ALLOCATION_TRACKING
     return 0;
-#else /* not GF_DISABLE_ALLOCATION_TRACKING */
+#else  /* not GF_DISABLE_ALLOCATION_TRACKING */
     if (!xl)
         return -1;
 
```

dmantipov commented:
/run regression

amarts reviewed (dismissed):
@dmantipov can you rebase one last time ? Good to merge this once rebase is done IMO. We surely can handle any major concerns as following PRs.

Commit:
libglusterfs: drop core infrastructure of memory pools

Separate memory allocation type tracking from core infrastructure of
memory pools and drop the latter alongside with '--disable-mempool'
configuration option, provide '--disable-allocation-tracking' to control
the former instead.

Signed-off-by: Dmitry Antipov <dantipov@cloudlinux.com>
Updates: #1000

gluster-ant commented:
CLANG-FORMAT FAILURE:
Before merging the patch, this diff needs to be considered for passing clang-format

```diff --git a/glusterfsd/src/glusterfsd.c b/glusterfsd/src/glusterfsd.c
index 5dca9f3f2..290e19dc1 100644
--- a/glusterfsd/src/glusterfsd.c
+++ b/glusterfsd/src/glusterfsd.c
@@ -1399,7 +1399,8 @@ parse_opts(int key, char *arg, struct argp_state *state)
         case ARGP_IO_ENGINE_KEY:
             cmd_args->io_engine = gf_strdup(arg);
             if (cmd_args->io_engine == NULL) {
-                argp_failure(state, -1, 0, "Failed to allocate memory for "
+                argp_failure(state, -1, 0,
+                             "Failed to allocate memory for "
                              "io-engine");
             }
             break;
@@ -2664,10 +2665,8 @@ out:
 int
 main(int argc, char *argv[])
 {
-    gf_io_handlers_t main_handlers = {
-        .setup = main_start,
-        .cleanup = main_terminate
-    };
+    gf_io_handlers_t main_handlers = {.setup = main_start,
+                                      .cleanup = main_terminate};
     glusterfs_ctx_t *ctx = NULL;
     int ret = -1;
     char cmdlinestr[PATH_MAX] = {
diff --git a/libglusterfs/src/xlator.c b/libglusterfs/src/xlator.c
index d00cf4872..c09ee2650 100644
--- a/libglusterfs/src/xlator.c
+++ b/libglusterfs/src/xlator.c
@@ -720,7 +720,7 @@ xlator_mem_acct_init(xlator_t *xl, int num_types)
 {
 #ifdef GF_DISABLE_ALLOCATION_TRACKING
     return 0;
-#else /* not GF_DISABLE_ALLOCATION_TRACKING */
+#else  /* not GF_DISABLE_ALLOCATION_TRACKING */
     if (!xl)
         return -1;
 
```

gluster-ant commented:
CLANG-FORMAT FAILURE:
Before merging the patch, this diff needs to be considered for passing clang-format

```diff --git a/glusterfsd/src/glusterfsd.c b/glusterfsd/src/glusterfsd.c
index 5dca9f3f2..290e19dc1 100644
--- a/glusterfsd/src/glusterfsd.c
+++ b/glusterfsd/src/glusterfsd.c
@@ -1399,7 +1399,8 @@ parse_opts(int key, char *arg, struct argp_state *state)
         case ARGP_IO_ENGINE_KEY:
             cmd_args->io_engine = gf_strdup(arg);
             if (cmd_args->io_engine == NULL) {
-                argp_failure(state, -1, 0, "Failed to allocate memory for "
+                argp_failure(state, -1, 0,
+                             "Failed to allocate memory for "
                              "io-engine");
             }
             break;
@@ -2664,10 +2665,8 @@ out:
 int
 main(int argc, char *argv[])
 {
-    gf_io_handlers_t main_handlers = {
-        .setup = main_start,
-        .cleanup = main_terminate
-    };
+    gf_io_handlers_t main_handlers = {.setup = main_start,
+                                      .cleanup = main_terminate};
     glusterfs_ctx_t *ctx = NULL;
     int ret = -1;
     char cmdlinestr[PATH_MAX] = {
diff --git a/libglusterfs/src/xlator.c b/libglusterfs/src/xlator.c
index d00cf4872..c09ee2650 100644
--- a/libglusterfs/src/xlator.c
+++ b/libglusterfs/src/xlator.c
@@ -720,7 +720,7 @@ xlator_mem_acct_init(xlator_t *xl, int num_types)
 {
 #ifdef GF_DISABLE_ALLOCATION_TRACKING
     return 0;
-#else /* not GF_DISABLE_ALLOCATION_TRACKING */
+#else  /* not GF_DISABLE_ALLOCATION_TRACKING */
     if (!xl)
         return -1;
 
```

xhernandez commented:
/run regression
